From 93cdf23c90fc4a0025ffcb5dc12b13f86551aa57 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Wed, 20 Feb 2013 23:10:29 +0300
Subject: [PATCH 439/675] libata: add R-Car SATA driver

Add Renesas R-Car on-chip 3Gbps SATA controller driver.

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
[Sergei: few bugs fixed, significant cleanup]
Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Signed-off-by: Jeff Garzik <jgarzik@redhat.com>
(cherry picked from commit 163cf81d266f017d718ef786fbbb6b2513ce7ec3)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 drivers/ata/sata_rcar.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/ata/sata_rcar.c b/drivers/ata/sata_rcar.c
index 5f3a23a..2e7cc60 100644
--- a/drivers/ata/sata_rcar.c
+++ b/drivers/ata/sata_rcar.c
@@ -818,7 +818,6 @@ static int sata_rcar_probe(struct platform_device *pdev)
 
 cleanup:
 	clk_disable(priv->clk);
-	clk_put(priv->clk);
 
 	return ret;
 }
@@ -837,7 +836,6 @@ static int sata_rcar_remove(struct platform_device *pdev)
 	iowrite32(0x7ff, priv->base + SATAINTMASK_REG);
 
 	clk_disable(priv->clk);
-	clk_put(priv->clk);
 
 	return 0;
 }
-- 
1.8.2

